### ISSUE TITLE
[Bug]: Inconsistent image outputs using VPred models + refiner using same seed/parameters

### ISSUE THREAD

**Issue by Elokhar-00** [Panchovix/stable-diffusion-webui-reForge#308]
### Checklist

- [x] The issue exists after disabling all extensions
- [x] The issue exists on a clean installation of webui
- [ ] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [x] The issue exists in the current version of the webui
- [x] The issue has not been reported before recently
- [ ] The issue has been reported before but has not been fixed yet

### What happened?

Images generated with the same seed and parameters when using  

- VPred base model with a refiner model (refiner can be VPred or EPS) 
OR 
- EPS base model with VPred refiner 

never give a consistent output for me on subsequently generating the same image. The images are similar, but never the same, sometimes only small details change, sometimes more prominent details. 

Conversely, images generated when using 

- VPred model without refiner 
OR 
- EPS base model + EPS refiner model
 
are exactly the same in subsequent generations, as expected.

Would be great to get this fixed for VPred models so the outputs stay consistent. Thank you in advance!

### Steps to reproduce the problem

Simple workflow overview that should showcase this:

- simple prompt "1girl, summer", no negatives
- Use same seed, batch count 1, batch size 1 (in case that matters)
- Base VPred model + refiner model enabled (refiner can be VPred or EPS)
- OR
- Base EPS model + VPred refiner model
- Hit generate
- Hit generate again without changing any settings

For me this generates two different images, and this is true for every subsequent generation with the same settings, they are never the same
If I now disable the refiner model and hit generate multiple times, the images stay consistent

One interesting thing I noticed is that if I close and reopen ReForge, and generate with the refiner model enabled, only the first generation of the image is consistent. 

I've tried several VPred models.

### What should have happened?

Generated images with VPred models when using a refiner should stay consistent when using the same parameters and seed, as is the case when using EPS base model + EPS refiner. 

### What browsers do you use to access the UI ?

Mozilla Firefox

### Sysinfo

[sysinfo-2025-03-08-10-41.json](https://github.com/user-attachments/files/19143124/sysinfo-2025-03-08-10-41.json)

### Console logs

```Shell
Already up to date.
venv "C:\ReForge\stable-diffusion-webui-reForge\venv\Scripts\Python.exe"
Python 3.12.9 (tags/v3.12.9:fdb8142, Feb  4 2025, 15:27:58) [MSC v.1942 64 bit (AMD64)]
Version: f1.3.0-v1.10.1RC-latest-1037-g525b1f51
Commit hash: 525b1f51a4e26c1f7415b04bf6ef3dbeaf168035
Legacy Preprocessor init warning: Unable to install insightface automatically. Please try run `pip install insightface` manually.
Launching Web UI with arguments: --pin-shared-memory --cuda-malloc --cuda-stream --ckpt-dir 'C:\A1111\stable-diffusion-webui\models\Stable-diffusion' --lora-dir 'C:\A1111\stable-diffusion-webui\models\Lora' --vae-dir 'C:\A1111\stable-diffusion-webui\models\VAE' --embeddings-dir 'C:\A1111\stable-diffusion-webui\embeddings'
Using cudaMallocAsync backend.
Total VRAM 24564 MB, total RAM 65220 MB
Set vram state to: NORMAL_VRAM
Always pin shared GPU memory
Device: cuda:0 NVIDIA GeForce RTX 4090 : cudaMallocAsync
VAE dtype: torch.bfloat16
CUDA Stream Activated:  True
Using pytorch attention
ControlNet preprocessor location: C:\ReForge\stable-diffusion-webui-reForge\models\ControlNetPreprocessor
C:\ReForge\stable-diffusion-webui-reForge\venv\Lib\site-packages\diffusers\models\transformers\transformer_2d.py:34: FutureWarning: `Transformer2DModelOutput` is deprecated and will be removed in version 1.0.0. Importing `Transformer2DModelOutput` from `diffusers.models.transformer_2d` is deprecated and this will be removed in a future version. Please use `from diffusers.models.modeling_outputs import Transformer2DModelOutput`, instead.
  deprecate("Transformer2DModelOutput", "1.0.0", deprecation_message)
Tag Autocomplete: Could not locate model-keyword extension, Lora trigger word completion will be limited to those added through the extra networks menu.
C:\ReForge\stable-diffusion-webui-reForge\modules\sd_models.py:524: FutureWarning: `torch.distributed.reduce_op` is deprecated, please use `torch.distributed.ReduceOp` instead
  if (isinstance(obj, torch.Tensor) and
Memory change: 0.15 MB (1.00 GB total)
Loading model naiXLVpred102d_final.safetensors [6f8dba7a27] (1 of 2)
Loading weights [6f8dba7a27] from C:\A1111\stable-diffusion-webui\models\Stable-diffusion\naiXLVpred102d_final.safetensors
2025-03-08 10:37:03,408 - ControlNet - INFO - ControlNet UI callback registered.
Running on local URL:  http://127.0.0.1:7860
model_type V_PREDICTION
UNet ADM Dimension 2816

To create a public link, set `share=True` in `launch()`.
Startup time: 10.9s (prepare environment: 2.1s, import torch: 3.8s, import gradio: 0.6s, setup paths: 0.8s, initialize shared: 0.1s, other imports: 0.3s, load scripts: 1.5s, create ui: 1.3s, gradio launch: 0.4s).
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
left over keys: dict_keys(['v_pred', 'ztsnr', 'cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids'])
Loading VAE weights specified in settings: C:\A1111\stable-diffusion-webui\models\VAE\liquid111vae_sdxl9745VAE.safetensors
Begin to load 1 model
[Memory Management] Current Free GPU Memory (MB) = 22968.74
[Memory Management] Model Memory (MB) = 2144.35
[Memory Management] Minimal Inference Memory (MB) = 1024.00
[Memory Management] Estimated Remaining GPU Memory (MB) = 19800.39
Moving model(s) has taken 0.40 seconds
Model naiXLVpred102d_final.safetensors [6f8dba7a27] loaded in 8.7s (memory cleanup: 0.3s, load weights from disk: 0.9s, forge load real models: 6.3s, load VAE: 0.4s, load textual inversion embeddings: 0.1s, calculate empty prompt: 0.6s).
C:\ReForge\stable-diffusion-webui-reForge\modules\sd_models.py:524: FutureWarning: `torch.distributed.reduce_op` is deprecated, please use `torch.distributed.ReduceOp` instead
  if (isinstance(obj, torch.Tensor) and
Memory change: 1.95 MB (15.24 GB total)
No new models to load
No new models to load
Begin to load 1 model
[Memory Management] Current Free GPU Memory (MB) = 19120.12
[Memory Management] Model Memory (MB) = 4897.09
[Memory Management] Minimal Inference Memory (MB) = 1024.00
[Memory Management] Estimated Remaining GPU Memory (MB) = 13199.04
Moving model(s) has taken 0.87 seconds
 50%|█████████████████████████████████████████                                         | 10/20 [00:01<00:01,  7.63it/s]GC pass 1: collected 6 objects█████████████████████████                                 | 10/20 [00:01<00:01,  8.31it/s]
Memory change: 5.51 MB (7.30 GB total)
Loading model pppanimixIL_75.safetensors [7fa37f3aa3] (2 of 2)
Loading weights [7fa37f3aa3] from C:\A1111\stable-diffusion-webui\models\Stable-diffusion\pppanimixIL_75.safetensors
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
left over keys: dict_keys(['cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids'])
Loading VAE weights specified in settings: C:\A1111\stable-diffusion-webui\models\VAE\liquid111vae_sdxl9745VAE.safetensors
Begin to load 1 model
[Memory Management] Current Free GPU Memory (MB) = 16219.97
[Memory Management] Model Memory (MB) = 2144.35
[Memory Management] Minimal Inference Memory (MB) = 1024.00
[Memory Management] Estimated Remaining GPU Memory (MB) = 13051.62
Moving model(s) has taken 0.44 seconds
Model pppanimixIL_75.safetensors [7fa37f3aa3] loaded in 8.0s (memory cleanup: 0.6s, load weights from disk: 0.9s, forge load real models: 5.4s, load VAE: 0.4s, load textual inversion embeddings: 0.1s, calculate empty prompt: 0.5s).
Memory change: 0.10 MB (16.32 GB total)
No new models to load
No new models to load
Begin to load 1 model
[Memory Management] Current Free GPU Memory (MB) = 14459.79
[Memory Management] Model Memory (MB) = 4897.09
[Memory Management] Minimal Inference Memory (MB) = 1024.00
[Memory Management] Estimated Remaining GPU Memory (MB) = 8538.71
Moving model(s) has taken 0.71 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:12<00:00,  1.60it/s]
Begin to load 1 model██████████████████████████████████████████████████████████████████| 20/20 [00:12<00:00,  4.07it/s]
[Memory Management] Current Free GPU Memory (MB) = 9564.41
[Memory Management] Model Memory (MB) = 159.56
[Memory Management] Minimal Inference Memory (MB) = 1024.00
[Memory Management] Estimated Remaining GPU Memory (MB) = 8380.86
Moving model(s) has taken 0.21 seconds
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  1.53it/s]
No new models to load██████████████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  4.07it/s]
No new models to load
Begin to load 1 model
Reuse 1 loaded models
[Memory Management] Current Free GPU Memory (MB) = 9345.59
[Memory Management] Model Memory (MB) = 0.00
[Memory Management] Minimal Inference Memory (MB) = 1024.00
[Memory Management] Estimated Remaining GPU Memory (MB) = 8321.59
Moving model(s) has taken 0.02 seconds
 50%|█████████████████████████████████████████                                         | 10/20 [00:01<00:01,  8.09it/s]No new models to load|█████████████████████████████████                                 | 10/20 [00:01<00:01,  8.60it/s]
No new models to load
No new models to load
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  7.58it/s]
No new models to load██████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  8.14it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  6.69it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  8.14it/s]
```

### Additional information

GPU drivers all up to date

**Comment by rashield938** [comment]
Exact same issue is happening to me too, so i had to revert it to commit 525b1f51a4e26c1f7415b04bf6ef3dbeaf168035 (March 7th), meanwhile it fixed the issue, but I would like to see it fixed in a future update.

**Comment by Elokhar-00** [comment]
Just wanted to give an update on this - on the latest commit at this time ( e97d988 ) this issue is gone, i.e., VPRED model + refiner model now gives consistent images on subsequent generations with the same parameters.
